File: source/com/barrybecker4/simulation/predprey/creatures/Rabbits.java
/** Copyright by Barry G. Becker, 2000-2011. Licensed under MIT License: http://www.opensource.org/licenses/MIT  */
package com.barrybecker4.simulation.predprey.creatures;

import java.awt.*;

/**
 * Everything we need to know about the current rabbit population.
 * Rabbits are the prey in the simulation.
 *
 * @author Barry Becker
 */
public class Rabbits extends Population {

    public static final Color COLOR = Color.BLUE;

    public static final int INITIAL_NUM_RABBITS = 100;
    private static final double RABBIT_BIRTH_RATE = 1.2;
    private static final double RABBIT_DEATH_RATE = 0.003;

    @Override
    public String getName() {
        return "Rabbit";
    }

    @Override
    public int getInitialPopulation() {
        return INITIAL_NUM_RABBITS;
    }
    @Override
    public double getInitialBirthRate() {
        return RABBIT_BIRTH_RATE;
    }
    @Override
    public double getInitialDeathRate() {
        return RABBIT_DEATH_RATE;
    }

    @Override
    public double getMaxDeathRate() {
        return 0.1;
    }
}

File: scala-source/com/barrybecker4/simulation/predprey/creatures/Rabbits.scala
// Copyright by Barry G. Becker, 2016-2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.simulation.predprey.creatures

import java.awt._


/**
  * Everything we need to know about the current rabbit population.
  * Rabbits are the prey in the simulation.
  * @author Barry Becker
  */
object Rabbits {
  val COLOR: Color = Color.BLUE
  val INITIAL_NUM_RABBITS = 100
  private val RABBIT_BIRTH_RATE = 1.2
  private val RABBIT_DEATH_RATE = 0.003
}

class Rabbits extends Population {
  override def getName = "Rabbit"
  override def getInitialPopulation: Int = Rabbits.INITIAL_NUM_RABBITS
  override def getInitialBirthRate: Double = Rabbits.RABBIT_BIRTH_RATE
  override def getInitialDeathRate: Double = Rabbits.RABBIT_DEATH_RATE
  override def getMaxDeathRate = 0.1
}
